### PR TITLE
feat(pci.project): instances creation reco tracking

### PR DESCRIPTION
**Pull request by selm3n**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #...TAPC-1312
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

### CODE REVIEW

**Review by oalkabouss — DISMISSED** [review]

**Comment by Eric-ciccotti** [comment]
manque la description de la MR et le ticket #feat plutôt que fix 😊

**Review by Eric-ciccotti — APPROVED** [review]

**Comment by y4nnL** [comment]
@selm3n can you squash the two commits into one as they have the same message ?

**Review by y4nnL — CHANGES_REQUESTED** [review]
@selm3n can you squash the two commits into one as they have the same message ?

File: packages/manager/modules/pci/src/components/project/flavor-billing/flavor-billing.component.js [hunk 1]
```diff
@@ -12,5 +12,6 @@ export default {
     addons: '<?',
     isLocalZone: '<',
     hourlyPriceInformation: '<',
+    onBillingTypeChange: '&?',
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/components/project/flavor-billing/flavor-billing.component.js:15]:
```suggestion
    onBillingTypeChange: '&',
```

File: packages/manager/modules/pci/src/components/project/flavor-billing/flavor-billing.controller.js [hunk 2]
```diff
@@ -33,6 +33,11 @@ export default class FlavorBillingController {
 
   selectBilling(monthlyBilling) {
     this.monthlyBilling = monthlyBilling;
+    if (this.onBillingTypeChange) {
+      this.onBillingTypeChange({
+        billing: monthlyBilling ? 'monthly' : 'hourly',
+      });
+    }
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/components/project/flavor-billing/flavor-billing.controller.js:40]:
```suggestion
    this.onBillingTypeChange({
      billing: monthlyBilling ? 'monthly' : 'hourly',
    });
```

File: packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.component.js [hunk 3]
```diff
@@ -22,5 +22,6 @@ export default {
     reload: '<',
     loadEnd: '&',
     displayLoader: '<?',
+    onTabChange: '&?',
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.component.js:25]:
```suggestion
    onTabChange: '&',
```

File: packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.controller.js [hunk 4]
```diff
@@ -152,6 +152,12 @@ export default class FlavorsListController {
     }
   }
 
+  onActiveChange(category) {
+    if (this.onTabChange) {
+      this.onTabChange({ category });
+    }
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.controller.js:158]:
```suggestion
    this.onTabChange({ category });
```

File: packages/manager/modules/pci/src/components/project/images-list/images-list.component.js [hunk 5]
```diff
@@ -15,6 +15,7 @@ export default {
     onChange: '&?',
     serviceName: '@',
     region: '<?',
+    onTabChange: '&?',
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/components/project/images-list/images-list.component.js:18]:
```suggestion
    onTabChange: '&',
```

File: packages/manager/modules/pci/src/components/project/images-list/images-list.controller.js [hunk 6]
```diff
@@ -176,8 +176,18 @@ export default class ImagesListController {
     }
   }
 
-  changeImageType() {
+  changeImageType(imageType) {
     this.imagesFromDistribution = [];
+
+    if (this.onTabChange) {
+      this.onTabChange({ type: imageType });
+    }
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/components/project/images-list/images-list.controller.js:184]:
```suggestion
    this.onTabChange({ type: imageType });
```

File: packages/manager/modules/pci/src/components/project/images-list/images-list.controller.js [hunk 7]
```diff
@@ -176,8 +176,18 @@ export default class ImagesListController {
...
+  }
+
+  changeApps(type) {
+    if (this.onTabChange) {
+      this.onTabChange({ type });
+    }
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/components/project/images-list/images-list.controller.js:190]:
```suggestion
    this.onTabChange({ type });
```

File: packages/manager/modules/pci/src/projects/new/payment/utils.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  if (parts.length === 2) {
+    return decodeURIComponent(
+      parts
+        .pop()
+        .split(';')
+        .shift(),
+    );
+  }
+  return '';
+}
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/projects/new/payment/utils.js:14]:
```suggestion
export function getVisitorId() {
  return document.cookie?.match(/clientSideUserId=([a-z0-9]+)?/i)?.[1] || '';
}
```

**Review by y4nnL — CHANGES_REQUESTED** [review]

**Review by Eric-ciccotti — APPROVED** [review]

**Review by oalkabouss — APPROVED** [review]

**Pull request closed without merging**